system-test/bigtable.ts

```diff
@@ -488,6 +491,83 @@ describe('Bigtable', () => {
       await table.get({autoCreate: true});
       await table.delete();
     });
+
+    it('should finish delivering a chunk every time a chunk is sent', async () => {
+      let rowCount = 0;
+      const chunkSize = 209;
+      const table = INSTANCE.table(generateId('table'));
+      const transformer = new Transform({
+        objectMode: true,
+        transform(
+          chunk: any,
+          _encoding: any,
+          callback: (err: any, data: any) => void
+        ) {
+          rowCount++;
+          setTimeout(() => {
+            callback(null, chunk);
+          }, 0);
+        },
+      });
+      const output = new Writable({
+        objectMode: true,
+        write(_chunk, _encoding, callback) {
+          callback();
+        },
+      });
+      const stream = streamEvents(
+        new PassThrough({
+          objectMode: true,
+        })
+      );
+      table.bigtable.request = (config?: any) => {
+        return stream as AbortableDuplex;
+      };
+      const readStream = table.createReadStream({});
+      function ascii(index: number) {
+        return String.fromCharCode(index);
+      }
+      function getChunk(index: number) {
+        return {
+          labels: [],
+          rowKey: Buffer.from(`a${ascii(index)}`),
+          familyName: {value: 'cf1'},
+          qualifier: {value: Buffer.from('a')},
+          timestampMicros: '12',
+          value: Buffer.from('a'),
+          valueSize: 0,
+          commitRow: true,
+          rowStatus: 'commitRow',
+        };
+      }
+      function pushChunks(numChunks: number) {
+        const chunks = [];
+        for (let i = 0; i < numChunks; i++) {
+          chunks.push(getChunk(i));
+        }
+        const data = {
+          chunks,
+          lastScannedRowKey: Buffer.from('a'),
+        };
+        stream.push(data);
+      }
+      setTimeout(() => {
```

Comment:
Change this to `process.nextTick`.

Reply:
Actually, maybe even emit each chunk at `nextTick` and then `end` at `nextTick` as well.

Reply:
I gave your former suggestion a try. For your latter suggestion do you mean something like this?

```
let chunksPushed = 0;
process.nextTick(() => {
    if (chunksPushed > 83) {
        pushChunks(chunkSize);
        chunksPushed++;
    } else {
        stream.emit('end');
    }
});
```